src/main/java/GulieStorage.java

```diff
@@ -0,0 +1,52 @@
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class GulieStorage {
+    private String savepath;
+    private GulieUi ui;
+
+    public GulieStorage(GulieUi ui, String savepath) {
+        this.savepath = savepath;
+        this.ui = ui;
+    }
+
+    public void save(GulieTasklist tasklist) throws GulieException {
+        try {
+            File save = new File(savepath);
+            save.mkdirs();
+            save.delete();
+            save.createNewFile();
+            FileWriter fw = new FileWriter(save);
+            for (Task t: tasklist) {
```

Comment:
Remember to surround your operator with space characters here!
```suggestion
            for (Task t : tasklist) {
```